esphome/cpp_helpers.py

```
@@ -100,9 +101,17 @@ async def setup_entity(var, config):
     """Set up generic properties of an Entity"""
     add(var.set_name(config[CONF_NAME]))
     if not config[CONF_NAME]:
-        add(var.set_object_id(sanitize(snake_case(CORE.friendly_name))))
+        object_id = sanitize(snake_case(CORE.friendly_name))
     else:
-        add(var.set_object_id(sanitize(snake_case(config[CONF_NAME]))))
+        object_id = sanitize(snake_case(config[CONF_NAME]))
+        #    If the result is empty or only underscores (e.g. from non-ASCII characters),fallback to a more reliable identifier
+        if not object_id or set(object_id) == {'_'}:
+            if CONF_ID in config:
+                object_id = sanitize(snake_case(str(config[CONF_ID])))
+            else:
+                from hashlib import md5
+                object_id = md5(config[CONF_NAME].encode("utf-8")).hexdigest()[:8]
```

wouldn't it be better to raise an error here?
ask the user to enter a valid id for this case

---

Thank you for the suggestion! You're absolutely right that raising an error would make the issue more visible and explicit.

However, doing so would unfortunately cause existing configurations from non-English users (who haven't set an `id:`) to fail entirely — even if their setups were previously working despite generating invalid `object_id`s.

To avoid breaking those configurations, a more graceful approach might be to **keep the MD5 fallback** for backward compatibility, but also **emit a warning** in the logs. This way, users are notified of the problem and encouraged to provide an explicit `id:`, without immediately breaking their devices.

---

What do you mean by their configurations already working?

I mean it is ok that the program compiles and runs, but with this change you are going to change an id which is ___ into a hash (and probably already break something ha side) so I would vote to fail here and give the user control of what name he would like to use. 

The fix is really simple from the user side, you can add in the message "Cannot determine an id from the given name. Please explicitly set a meaningful id: for this entity"

---

Do you mean changing it to something like this?

```python
        object_id = sanitize(snake_case(str(config[CONF_ID])))
    else:
        _LOGGER.error(f"The name '{config[CONF_NAME]}' is invalid after sanitization.")
        raise cv.Invalid(
            f"Cannot determine an object_id from the given name '{config[CONF_NAME]}'. "
            f"Please explicitly set a meaningful id: for this entity."
        )
```

As you mentioned, falling back to an MD5 hash might have already broken something on the HA side — that's something I hadn't considered before. Thanks for the clear and helpful explanation!

---

Yes I meant that. 
May be the error log is not necessary 

I'm not sure if this change will affect the desire to make the entity names translatable. But for me it is good enough